packages/truffle-db/src/workspace/test/index.ts

```diff
@@ -249,274 +276,221 @@ mutation AddCompilation($compilerName: String!, $compilerVersion: String!, $sour
         name
       }
       sources {
-        id
+        contents
+      }
+      contracts {
+        source {
+          contents
+          sourcePath
+        }
+        ast {
+          json
+        }
+        name
       }
     }
   }
 }`
 
-it("adds contracts", async () => {
-  const workspace = new Workspace();
-  const id = soliditySha3(Migrations.contractName, soliditySha3(Migrations.source, Migrations.sourcePath))
-  
-  const variables = {
-    contractName: Migrations.contractName,
-    sourceId: soliditySha3(Migrations.source, Migrations.sourcePath),
-    id: id
-  }
-
-  // add contracts
-  {
-    const result = await graphql(
-      schema, AddContracts, null, { workspace }, variables
-    );
-
-    const { data } = result;
-  
-    expect(data).toHaveProperty("contractsAdd");
-
-    const { contractsAdd } = data;
-    expect(contractsAdd).toHaveProperty("contracts");
+describe("Compilation", () => {
+  it("adds compilation", async () => {
+    const client = new WorkspaceClient();
+
+    const variables = {
+      compilerName: Migrations.compiler.name,
+      compilerVersion: Migrations.compiler.version,
+      sourceId: sourceId,
+      id: compilationId,
```

Comment:
And here, `compilationId` would be `expectedCompilationId` instead of a variable.

Reply:
And instead of passing pre-generated `sourceId`, I think we should add a `before("add source", ...)` block inside `describe("Compilation", ...)`, where we assign `sourceId` to the _actual_ ID that gets generated for the source, rather than pre-generating it. Since we're testing Compilation here, not Source.